network/peer.go

```diff
@@ -784,3 +929,135 @@ func (p *peer) discardMyIP() {
 	}
 	p.Close()
 }
+
+// Read a single message from the connection.
+func (p *peer) readMsg() (Msg, error) {
+	pendingBuffer := wrappers.Packer{}
+	readBuffer, err := p.readFull(4)
+	if err != nil {
+		return nil, err
+	}
+	if len(readBuffer) != 4 {
+		return nil, fmt.Errorf("read buffer failed")
+	}
+
+	pendingBuffer.Bytes = append(pendingBuffer.Bytes, readBuffer...)
+
+	// lets figure out the size of the message..
+	size := pendingBuffer.UnpackInt()
+
+	// now lets read the message.
+	readBuffer, err = p.readFull(size)
+	if err != nil {
+		return nil, err
+	}
+	if uint32(len(readBuffer)) != size {
+		return nil, fmt.Errorf("read buffer failed")
+	}
+
+	pendingBuffer.Bytes = append(pendingBuffer.Bytes, readBuffer...)
+
+	// reset the offset after the size read.
+	pendingBuffer.Offset = 0
+
+	// unpack the bytes or error out...
+	msgBytes := pendingBuffer.UnpackBytes()
+	if pendingBuffer.Errored() {
+		return nil, pendingBuffer.Err
+	}
+
+	var msg Msg
+	msg, err = p.net.b.Parse(msgBytes)
```

Comment:
Is there any reason not to use the syntax `msg, err := p.net.b.Parse(msgBytes)` here to save a line?

Reply:
at least in goland identifies 'err' is previously defined, and is being overwritten with ':='....

![image](https://user-images.githubusercontent.com/67973730/95901148-e8e0cc00-0d60-11eb-9a35-582076cd367c.png)

Just trying to have clean IDE...

Reply:
fixed..